src/App.tsx

```diff
@@ -38,58 +41,104 @@ export function getReorderedGoods(
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
-  return visibleGoods;
+  if (sortType !== 0) { // Check if sortType is not default 0 (NONE value)
+    if (sortType === 1) { // sortType[1] === ALPHABET
+      visibleGoods.sort(
+        (goodOne, goodAnother) => goodOne.localeCompare(goodAnother),
+      );
+    } else { // sortType[2] === LENGTH
+      visibleGoods.sort(
+        (goodOne, goodAnother) => goodOne.length - goodAnother.length,
+      );
+    }
+  }
+
+  return !isReversed
+    ? visibleGoods
+    : visibleGoods.reverse();
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const {
+      isReversed: isRev,
+      sortType: sortBy,
+    } = this.state;
+
+    const options = {
+      sortType: sortBy || SortType.NONE,
+      isReversed: isRev || false,
+    };
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              classNames('button', 'is-info', { 'is-light': sortBy !== 1 })
+            }
+            onClick={() => this.setState({ sortType: SortType.ALPHABET })}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames('button', 'is-success', { 'is-light': sortBy !== 2 })
+            }
+            onClick={() => this.setState({ sortType: SortType.LENGTH })}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames('button', 'is-warning', { 'is-light': !isRev })
+            }
+            onClick={() => this.setState(
+              (prevState) => ({ isReversed: !prevState.isReversed }),
+            )}
+          >
+            Reverse
+          </button>
+
+          {(isRev || sortBy !== 0)
```

Comment:
Create a constant for this condition

Reply:
initialized `reordered` state property with default false value